### PR TITLE
Fuzzing: Compare wasm2js to the interpreter

### PR DESCRIPTION
**Pull request by kripken**
Comparing to the interpreter, and not just wasm2js to itself (which we've
done on the same file before and after opts) ensures wasm2js has the right
semantics.

To do this, we need to make sure the wasm doesn't contain things where
wasm2js semantics diverge from normal wasm, which includes:

* Legalize so that there are no i64 exports.
* Remove operations JS can't handle with full precision, like i64 -> f32.
* Force all loads/stores to be 1-byte, as unexpectedly-unaligned operations
  fail in wasm2js.

This also requires ignoring subnormals when comparing between JS
VMs and the interpreter.

### CODE REVIEW

File: scripts/fuzz_opt.py [hunk 1]
```diff
@@ -187,14 +187,16 @@ def numbers_are_close_enough(x, y):
     # float() on the strings will handle many minor differences, like
     # float('1.0') == float('1') , float('inf') == float('Infinity'), etc.
     try:
-        return float(x) == float(y)
+        fx = float(x)
+        fy = float(y)
+        return fx == fy
```
Comment (tlively) [inline scripts/fuzz_opt.py:192]: Why this change?

File: scripts/fuzz_opt.py [hunk 2]
```diff
@@ -532,18 +559,53 @@ def handle_pair(self, input, before_wasm, after_wasm, opts):
...
+                    # subnormals in full detail while other VMs do not, and vice
+                    # versa. Ignore such really tiny numbers.
+                    if is_basically_zero(x):
+                        x = 0
+                except ValueError:
+                    # not a floating-point number, nothing to do
+                    pass
+                return ' => ' + str(x)
+
+            return re.sub(r' => (-?[\d+-.e\-+]+)', fix_number, x)
```
Comment (tlively) [inline scripts/fuzz_opt.py:600]: What black magic is this!?

File: scripts/fuzz_opt.py (hunk 1 above)
Comment (kripken) [reply scripts/fuzz_opt.py:192]: It can help when debugging as then it an exception is thrown you can see on which line it happens. But maybe not worth it just for that, as it is less clear, I'll remove it.

File: scripts/fuzz_opt.py (hunk 2 above)
Comment (kripken) [reply scripts/fuzz_opt.py:600]: "simple" regex to detect a subset of floating point numbers... The `=>` is unrelated, I'll clarify that in a comment.
